### ISSUE TITLE
Metadata not being generated for intercepting routes but generated for parallel routes

### ISSUE THREAD

**Issue by EmmanuelVictor62** [vercel/next.js#68562]
### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/intercepting-route-metadata-sample-mglmyj

### To Reproduce

1. Start the development 
2. On the default page shown, click on the any of the category link
3. On the intercepted route shown, notice metadata is not generated
4. On the parallel route, metadata is generated

### Current vs. Expected behavior

Following the examples in the doc, I expected that a metadata should be generated for the intercepting route but this does not get generated, I do not want to use the client side approach for proper SEO

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Pro
  Available memory (MB): 8073
  Available CPU cores: 4
Binaries:
  Node: 20.5.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.2.5 // Latest available version is detected (14.2.5).
  eslint-config-next: 14.1.2
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Metadata, Parallel & Intercepting Routes

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local), next start (local)

**Comment by dir** [comment]
I tried to use parallel routes, like many others. Unfortunately I ran into the exact same issue you're having and some. Strange issues with metadata, weird rendering/loading state bugs. 

This isn't really an answer, but I'd honestly say that parallel routes in their current implementation are just not a complete product.

**Comment by EmmanuelVictor62** [comment]
You're absolutely right. Parallel routes and Intercepting routes in their current state are still a bit rough around the edges. For me, it is not production ready yet🤧.

Hopefully, the Next.js team will work on this feature and make it better, great work so far though.
